### PR TITLE
Move freestyle rpc to freestyle free rpc

### LINKED ISSUE #469: SPIKE: Investigate about fs2-grpc library
https://github.com/fiadliel/fs2-grpc

We need to know how to build a client and a server and what configuration options provides.

Dependency: https://mvnrepository.com/artifact/org.lyranthe.fs2-grpc/java-runtime_2.12/

### PR DESCRIPTION
**Pull request by AdrianRaFo**
Fixes https://github.com/frees-io/freestyle/issues/469

### CODE REVIEW

**Comment by raulraja** [comment]
The RPC and Cassandra lib does not need to be aligned, just have it's macros adjusted if any so it compiles with the new core.

File: build.sbt [hunk 1]
```diff
@@ -28,7 +28,7 @@ lazy val common = project
   .settings(moduleName := "frees-rpc-common")
   .settings(scalacOptions := Seq("-deprecation", "-encoding", "UTF-8", "-feature", "-unchecked"))
 
-lazy val freesV = "0.4.6"
+lazy val freesV = "0.4.7"
```
Comment (juanpedromoreno) [inline build.sbt]: I guess the new version of Freestyle core will be 0.5.0, due the breaking changes.

File: rpc/src/main/scala/free/RPCAsyncImplicits.scala [hunk 2]
```diff
@@ -14,12 +14,12 @@
  * limitations under the License.
  */
 
-package freestyle
+package freestyle.free
 package rpc
```
Comment (juanpedromoreno) [inline rpc/src/main/scala/free/RPCAsyncImplicits.scala]: Not needed.

**Comment by raulraja** [comment]
Looks like the build needs to be updated to not depend on the tagless module anymore.

**Comment by juanpedromoreno** [comment]
As we had several conflicts (including macros code) I preferred to do it from a fresh branch cut off from the master branch here: https://github.com/frees-io/freestyle-rpc/pull/99

Thanks @AdrianRaFo !

**Pull request closed without merging**